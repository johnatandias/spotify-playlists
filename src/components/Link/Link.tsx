import React from 'react';

interface Link {
  href: string,
  className?: string
}

export const Link: React.FC<Link> = ({ href, className, children }) => (
  <a href={href} className={className}>
    {children}
  </a>
);
