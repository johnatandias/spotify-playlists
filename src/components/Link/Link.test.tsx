import React from "react";
import { shallow } from 'enzyme';
import { Link } from './Link';

describe('Link component', () => {
  it('Renders without crashing', () => {
    const wrapper = shallow(<Link href="www.google.com" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Renders with classname', () => {
    const className = 'my-class';
    const wrapper = shallow(<Link href="www.google.com" className={className} />);
    expect(wrapper.hasClass(className)).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });

  it('Renders with children', () => {
    const ChildrenComponent = () => <div>My children component</div>;
    const wrapper = shallow(
      <Link href="www.google.com">
        <ChildrenComponent />
      </Link>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
