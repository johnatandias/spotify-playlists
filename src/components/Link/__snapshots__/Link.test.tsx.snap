// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component Renders with children 1`] = `ShallowWrapper {}`;

exports[`Link component Renders with classname 1`] = `ShallowWrapper {}`;

exports[`Link component Renders without crashing 1`] = `ShallowWrapper {}`;
